doc/howtos/pull-request-guidelines.rst

```diff
@@ -0,0 +1,161 @@
+  information (see next point).
+* Searching for problems and how they were solved: ``git log --grep`` can be
+  useful to find commits again.
+* Commit author: when including something written by others (like patches
+  or complete files), try to submit that in a separate commit with the
+  original author set as author of the commit. This is important for tracking
+  the origin of code in Ostro OS, which partly is done by automated tooling.
+  When multiple people authored the change, use the upstream project's
+  mailing list as author. Alternatively, using ``git am`` format for patches
+  can also be used to document the author of a code patch (see next section).
+
+
+Content of a Pull Request
+=========================
+
+When a change affects user-visible behavior, make sure that relevant
+user documentation gets updated, ideally in the same commit (only
+works when submitting to ``meta-ostro``, which is where the ``doc``
+directory is maintained). If that is not possible or not desirable at
+the time, include enough information in the change or commit message
+for a technical writer to update the Ostro OS user documentation
+later.
+
+Avoid unnecessary changes, like arbitrarily reformatting unrelated code.
+
+When adding patches inside a PR, make sure that:
+
+* It is really necessary: can the problem be fixed upstream first? Are
+  there other solutions that are easier to maintain?
+* Document the patch status by adding a ``Upstream-Status`` line in
+  the `patch header`_. "Pending" patches are discouraged because they
+  contribute to the technical debt in Ostro OS.
+* The author of the patch is clear, either by using ``git am``
+  format with an ``Author:`` line, adding such a line manually to
+  the patch header, or by committing the entire file with the author
+  of the commit set to the author of the patch.
+
+.. _patch header: http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations
+
+Adding or removing layers needs to be done in two separate PRs:
+
+* In ``ostro-os``: update combo-layer.conf and include the initial
+  import of the new component in the PR
+* In ``meta-ostro`` update:
+ 
+  * ``.gitignore`` - imported files must not be ignored
+  * ``README.rst`` - the list of layers
+  * ``meta-ostro/conf/bblayers.conf.sample`` - add the layer and increase
+    ``LCONF_VERSION`` by one
+  * ``meta-ostro/conf/layer.conf`` - increase ``LAYER_CONF_VERSION``
+    by one. It must match ``LCONF_VERSION``.
+
+The layer config file versioning ensures that developers who update to
+a new revision of ``ostro-os`` where the new layer was added are told
+to update their local ``bblayers.conf`` when invoking bitbake in older
+build directories
+
+Similarly, when making changes to ``local.conf.sample`` that
+developers using an older copy of the file should be aware of,
+increase both ``LOCALCONF_VERSION`` in ``meta-ostro/conf/layer.conf``
+and ``CONF_VERSION`` in ``meta-ostro/conf/local.conf.sample``.
+
```

Comment:
Actually, the "similarly" covers that, because the implication was already explained for the bblayers.conf file. I just shortened the entire paragraph as suggested in the earlier comment, without adding this additional sentence.